mods/ra/maps/soviet-02a/soviet02a.lua

```diff
@@ -0,0 +1,170 @@
+
+		Utils.Do(FleeingUnits, function(unit)
+			unit.Move(waypoint4.Location)
+		end)
+		Follower.AttackMove(waypoint4.Location)
+	end)
+
+	-- To make it look more smooth we will blow up the bridge when the barrel closest to it blows up
+	Trigger.OnAnyKilled({ BridgeBarrel1, BridgeBarrel2 }, function()
+		-- Destroy the bridge
+		if not bridgepart1.IsDead then
+			bridgepart1.Kill()
+		end
+		if not bridgepart2.IsDead then
+			bridgepart2.Kill()
+		end
+
+		-- Move units towards the oilpump
+		--[[Utils.Do(FleeingUnits, function(unit)
+			if not unit.IsDead then
+				unit.Move(waypoint53.Location)
+			end
+		end)]]
+	end)
+
+	-- If player passes over the bridge, blow up the barrel and destroy the bridge
+	Trigger.OnEnteredFootprint(PassingBridgeLocation, function(unit, id)
+		if unit.Owner == player then
+			Trigger.RemoveFootprintTrigger(id)
+
+			-- Also don't if the bridge is already dead
+			if bridgepart1.IsDead and bridgepart2.IsDead then
+				return
+			end
+
+			-- Don't "shoot" at the barrels if there is noone to shoot
+			if not FleeingUnits[1].IsDead then
+				FleeingUnits[1].Attack(Barrel)
+			elseif not FleeingUnits[2].IsDead then
+				FleeingUnits[2].Attack(Barrel)
+			end
+		end
+	end)
+
+	-- Four infantry from the small island move towards the USSR command center and attack it after 24 Seconds
+	Trigger.AfterDelay(DateTime.Seconds(24),  function()
+		Utils.Do(CmdAtk, function(unit)
+			unit.AttackMove(AttackWaypoint1.Location)
+			Trigger.OnIdle(unit, unit.Hunt)
+			--Trigger.OnIdle(unit, function() unit.AttackMove(AttackWaypoint1.Location) end)
+		end)
+
+		-- On reaching waypoint 7 remove the trigger to keep moving or substitue with new instruction
+		--[[Trigger.OnEnteredFootprint({ AttackWaypoint1.Location }, function(unit, id)
+			Trigger.RemoveFootprintTrigger(id)
+
+			Utils.Do(CmdAtk, function(unit)
+				if not unit.IsDead then
+					--remove the attackmove trigger
+					--Trigger.Clear(unit, "OnIdle")
+
+					--TODO: Order the unit to attack the commandcenter, there is currently no command to do so
+					Trigger.OnIdle(unit, function() unit.Hunt() end)
+				end
+			end)
+		end)]]
+	end)
+
+	-- Start hunting
+	Hunter4.AttackMove(AttackWaypoint2.Location) -- Move the unit in the correct direction first
+	Utils.Do(HuntingUnits, function(unit)
+		Trigger.OnIdle(unit, unit.Hunt)
+	end)
+
+	-- When destroying the allied radar dome or the refinery drop 2 badgers with 5 grenadiers each
+	Trigger.OnAnyKilled({ AlliedDome, AlliedProc }, function()
+		local powerproxy = Actor.Create("powerproxy.paratroopers", true, { Owner = player })
+		powerproxy.SendParatroopers(ParadropLZ.CenterPosition, false, Facing.South)
+		powerproxy.SendParatroopers(ParadropLZ.CenterPosition, false, Facing.SouthEast)
+		powerproxy.Destroy()
+	end)
+
+	-- When passing the map coordinates in CameraTriggerArea reveal the allied base
+	--[[Trigger.OnEnteredFootprint(CameraTriggerArea, function(unit, id)
```

Comment:
I don't think a camera is necessary, as the player probably has the entire base already revealed using yaks.